### PR TITLE
ci: tighten run requirements, apply formatting

### PR DESCRIPTION
**Pull request by amaanq**
* For example, a typo fix doesn't need every action to run
* Added a \n above the parsers list in the README as formatting specs indicate it should be there, and this will alleviate people using autoformatters having to remove that

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -2,7 +2,11 @@ name: Linting and style checking
 
 on:
   push:
+    paths:
+      - "lua/**"
```
Comment (lucario387) [inline .github/workflows/lint.yml:6]: Does this mean this doesn't lint tests files anymore?

File: .github/mergify.yml [hunk 2]
```diff
@@ -12,9 +12,9 @@ pull_request_rules:
   - name: Prepare for merge
     conditions:
       - and:
-        - "-draft"
-        - "#approved-reviews-by=1"
-        - "#review-requested=0"
+          - "-draft"
+          - "#approved-reviews-by=1"
+          - "#review-requested=0"
```
Comment (lucario387) [inline .github/mergify.yml:17]: Seems like formatting error to me. Shouldn't all `- ...` be indented more here?

Comment (amaanq) [reply .github/mergify.yml:17]: No? But I think it comes from prettier's formatting

Comment (lucario387) [reply .github/mergify.yml:17]: Ok then :+1:

**Comment by amaanq** [comment]
Only issue is CI still "requires" tests even if they're "not needed" depending on the files changes - not sure how I'd fix that

**Comment by lucario387** [comment]
> Only issue is CI still "requires" tests even if they're "not needed" depending on the files changes - not sure how I'd fix that

Probably git diff from within the workflow and check for the diffs -> from that get the changes?

File: .github/mergify.yml [hunk 3]
```diff
@@ -25,13 +25,13 @@ pull_request_rules:
...
+          - or:
+              - "#approved-reviews-by>=2"
+              - and:
+                  - "#approved-reviews-by=1"
+                  - "updated-at>=1 day ago"
+          - "-draft"
+          - "#review-requested=0"
     actions:
       merge:
         method: rebase
```
Comment (theHamsta) [inline .github/mergify.yml:37]: We disabled mergify. How about removing this file?

File: .github/workflows/test-queries.yml [hunk 4]
```diff
@@ -4,9 +4,15 @@ on:
   push:
     branches:
       - "master"
+    paths:
+      - "lockfile.json"
+      - "queries/**"
```
Comment (theHamsta) [inline .github/workflows/test-queries.yml:9]: We already have a cache for regenerating the parsers, changes in the build/install logic should also trigger building and tests.

It's very tricky to discover bugs when a important file of the cache was forgotten. Not sure if the paths could be inverted to only apply to your documentation usecase

Comment (theHamsta) [reply .github/workflows/test-queries.yml:9]: Same holds for changes to parser.lua which add a new parser. Would try to use the same pattern as for the parser build cache plus queries

**Pull request closed without merging**